File: Difficulty: Basic/Prime Number/prime-number.java
//{ Driver Code Starts
import java.io.*;
import java.util.*;

class GFG
{
    public static void main(String args[])throws IOException
    {
        BufferedReader read = new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(read.readLine());
        while(t-- > 0)
        {
            int N = Integer.parseInt(read.readLine());
            Solution ob = new Solution();
            System.out.println(ob.isPrime(N));
        }
    }
}
// } Driver Code Ends


class Solution{
    static int isPrime(int N){
        // code here
        if(N <= 1)  return 0;
        if(N <= 3)  return 1;
        
        return primeHelper(N, 2);
    }
    
    static int primeHelper(int n, int i){
        if(i*i > n) return 1;
        
        if(n % i == 0)  return 0;
        
        return primeHelper(n, i+1);
    }
}

File: Difficulty: Easy/Prime Number/prime-number.py
class Solution:
    def isPrime(self, n):
        # code here
        if n <= 1:
            return False
            
        if n == 2:
            return True
            
        if n % 2 == 0:
            return False
        
        
        for i in range(3, int(n**0.5)+1, 2):
            if n % i == 0:
                return False
                
    
        return True